jax/ad_util.py

```diff
@@ -43,13 +44,19 @@ def add_impl(xs, ys):
 def add_abstract(xs, ys):
   return lattice_join(xs, ys)
 
+def float0s_like(a):
+  return core.Float0Array(a.shape)
+
 jaxval_zeros_likers: Dict[type, Array] = {}
+jaxval_zeros_likers[dtypes.dtype(core.Float0Array)] = float0s_like
+jaxval_zeros_likers[dtypes.dtype(int)] = float0s_like
 
 def zeros_like_aval(aval):
   return aval_zeros_likers[type(aval)](aval)
 
 aval_zeros_likers: Dict[type, Array] = {}
 aval_zeros_likers[AbstractUnit] = lambda _: unit
+aval_zeros_likers[core.Float0Array] = lambda aval: aval
```

Comment:
I think the keys in this table should be AbstractValue subclasses (and the type annotation should be more precise). Since float0 arrays reuse existing abstract values rather than their own, we probably don't need to extend the table.

```suggestion
```